test/libsolidity/SemanticTest.cpp

```diff
@@ -300,6 +336,31 @@ TestCase::TestResult SemanticTest::runTest(
 	return TestResult::Success;
 }
 
+bool SemanticTest::checkGasCostExpectation(TestFunctionCall& io_test, bool _compileViaYul) const
+{
+	string setting =
+		(_compileViaYul ? "ir"s : "legacy"s) +
+		(m_optimiserSettings == OptimiserSettings::full() ? "Optimized" : "");
+
+	// We don't check gas if enforce gas cost is not active
+	// or in case we run abiencoderv1
```

Comment:
I'd add an assertion about ABI encoder here. It's good that it's a parameter check now but, well, it would also be good to have a sanity check here :)

Reply:
Added assertion